lamindb/models/transform.py

```diff
@@ -328,13 +336,87 @@ def __init__(
         )
 
     @property
+    @deprecated
     def name(self) -> str:
         """Name of the transform.
 
         Splits `key` on `/` and returns the last element.
         """
         return self.key.split("/")[-1]
 
+    @classmethod
+    def from_git(
+        cls,
+        url: str,
+        path: str,
+        key: str | None = None,
+        version: str | None = None,
+        entrypoint: str | None = None,
+        branch: str | None = None,
+    ) -> Transform:
+        """Create a transform record from a path in a git repository.
+
+        This is particularly useful for Nextflow pipelines.
+
+        Examples:
+
+            Create from a Nextflow repo and auto-infer the commit hash from its latest version::
+
+                transform = ln.Transform.from_git(
+                    url="https://github.com/openproblems-bio/task_batch_integration",
+                    path="main.nf"
+                ).save()
+
+            Create from a Nextflow repo and checkout a specific version::
+
+                transform = ln.Transform.from_git(
+                    url="https://github.com/openproblems-bio/task_batch_integration",
+                    path="main.nf",
+                    version="v2.0.0"
+                ).save()
+
+            Create a sliding transform from a Nextflow repo's `dev` branch::
+
+                transform = ln.Transform.from_git(
+                    url="https://github.com/openproblems-bio/task_batch_integration",
+                    path="main.nf",
+                    branch="dev",
+                    version="dev",
+                ).save()
+
+        """
+        from ..core._sync_git import get_and_validate_git_metadata
+
+        url, commit_hash = get_and_validate_git_metadata(url, path, version, branch)
+        if key is None:
+            key = url.split("/")[-2] + "/" + url.split("/")[-1].replace(".git", "")
+            and_path = ""
+            if "main" in path:
+                key += "/" + path
+                and_path = "& path"
+            logger.important(f"inferred key '{key}' from url {and_path}")
+        source_code = f"repo: {url}\npath: {path}"
+        if entrypoint is not None:
+            source_code += f"\nentrypoint: {entrypoint}"
+        if branch is not None and version == branch:
+            # sliding transform, no defined source code state
+            source_code += f"\nbranch: {branch}"
+            reference = None
+            reference_type = None
+        else:
+            # regular transform, defined source code state
+            source_code += f"\ncommit: {commit_hash}"
+            reference = f"{url}/blob/{commit_hash}/{path}"
```

Comment:
This will only work for GitHub/Gitea/Gogs repositories, not for GitLab, BitBucket or Azure DevOps.